cscs-checks/system/nvidia/nvidia_smi_check.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+def sanity_evaluation(regex, regex_group, output, refvalue, field_msg, num_nodes, nodelist):
```

Comment:
Most or all the logic here can go in a pre-sanity hook that would also set the `sanity_patterns`.